src/OpenLoco/src/Vehicles/OrderManager.cpp

```diff
@@ -98,6 +109,39 @@ namespace OpenLoco::Vehicles::OrderManager
         }
     }
 
+    bool spaceLeftInGlobalOrderTableForOrder(const Order* order)
+    {
+        return numOrders() + kOrderSizes[enumValue(order->getType())] <= Limits::kMaxOrders;
+    }
+
+    bool spaceLeftInVehicleOrderTableForOrder(VehicleHead* head, const Order* order)
+    {
+        return head->sizeOfOrderTable + kOrderSizes[enumValue(order->getType())] < Limits::kMaxOrdersPerVehicle;
+    }
+
+    // 0x004704AB
+    void insertOrder(VehicleHead* head, uint16_t orderOffset, const Order* order)
+    {
+        auto insOrderLength = kOrderSizes[enumValue(order->getType())];
+
+        // Shift current order to compensate for the new order?
+        if (head->currentOrder >= orderOffset)
+        {
+            head->currentOrder += insOrderLength;
```

Comment:
Vanilla had a check for if the sizeofOrderTable == 1 don't increment. This because an empty order table is size 1 and in that case you want current order to point at the new order not the endOrder.

Reply:
May as well add a note that 1 length tables shouldn't adjust the order as we want it to now point at the first valid order.